samples/nrf9160/nrf_cloud_rest_cell_pos/src/main.c

```diff
@@ -0,0 +1,580 @@
+	} while (err != 0);
+
+	LOG_INF("Network time obtained");
+}
+
+#if defined(CONFIG_REST_CELL_POS_DO_JITP)
+static void request_jitp(void)
+{
+	int err;
+
+	k_sem_reset(&button_press_sem);
+
+	LOG_INF("---> Press button %d to request just-in-time provisioning (JITP)",
+		BTN_NUM);
+	LOG_INF("     This only needs to be done once per device");
+	LOG_INF("     Waiting %d seconds...", JITP_REQ_WAIT_SEC);
+
+	err = k_sem_take(&button_press_sem, K_SECONDS(JITP_REQ_WAIT_SEC));
+	if (err == 0) {
+		jitp_requested = true;
+		LOG_INF("JITP will be performed after network connection is obtained");
+		/* Enable the location card for a newly JITP'd device */
+		location_card_enable = true;
+	} else {
+		if (err != -EAGAIN) {
+			LOG_ERR("k_sem_take error: %d", err);
+		}
+
+		LOG_INF("JITP will be skipped");
+	}
+}
+
+static void do_jitp(void)
+{
+	int err = 0;
+
+	LOG_INF("Performing JITP...");
+	err = nrf_cloud_rest_jitp(CONFIG_NRF_CLOUD_SEC_TAG);
+
+	if (err == 0) {
+		LOG_INF("Waiting 30s for cloud provisioning to complete...");
+		k_sleep(K_SECONDS(30));
+		k_sem_reset(&button_press_sem);
+		LOG_INF("Associate device with nRF Cloud account and press button %d when complete",
+			BTN_NUM);
+		(void)k_sem_take(&button_press_sem, K_FOREVER);
+	} else if (err == 1) {
+		LOG_INF("Device already provisioned");
+	} else {
+		LOG_ERR("Device provisioning failed");
+	}
+}
+#endif
+
+#define REQ_CARD "---> Press button %d to enable the location card for this device on nrfcloud.com"
+static void request_location_card_enable(void)
+{
+	if (!location_card_enable) {
+		int err;
+
+		k_sem_reset(&button_press_sem);
+
+		LOG_INF(REQ_CARD, BTN_NUM);
+		LOG_INF("     This only needs to be done once per device");
+		LOG_INF("     Waiting %d seconds...", UI_REQ_WAIT_SEC);
+
+		err = k_sem_take(&button_press_sem, K_SECONDS(UI_REQ_WAIT_SEC));
+
+		if (err == 0) {
+			location_card_enable = true;
+		} else {
+			if (err != -EAGAIN) {
+				LOG_ERR("k_sem_take error: %d", err);
+			}
+		}
+	}
+
+	if (location_card_enable) {
+		LOG_INF("Location card will be enabled after network connection is obtained");
+	} else {
+		LOG_INF("Location card will not be enabled");
+	}
+}
+
+static void do_location_card_enable(void)
+{
+	int err;
+	struct nrf_cloud_svc_info_ui ui = {
+		.gps = 1
+	};
+	struct nrf_cloud_svc_info svc_inf = {
+		.fota = NULL,
+		.ui = &ui
+	};
+
+	/* Keep the connection alive so it can be used by the initial cellular
+	 * positioning request.
+	 */
+	rest_ctx.keep_alive = true;
+
+	LOG_INF("Enabling location card...");
+	err = nrf_cloud_rest_shadow_service_info_update(&rest_ctx, device_id, &svc_inf);
+	if (err) {
+		LOG_ERR("Failed to enable location card, error: %d", err);
+	} else {
+		LOG_INF("Location card enabled in device shadow");
+	}
+
+	/* Set keep alive to false so the connection is closed after the initial
+	 * positioning request. The subsequent request may not occur immediately.
+	 */
+	rest_ctx.keep_alive = false;
```

Comment:
Doesn't this pretty quickly just cancel out what you do on line 341?  When would the initial cellular positioning request have time to occur between these two lines?

Reply:
this is just a flag, the connection will be kept alive after `nrf_cloud_rest_shadow_service_info_update` is complete and it will be closed after the initial cell pos request.

Reply:
OK

Reply:
The code might read a little easier if keep_alive is just kept false the whole time, if the downside of making two separate connections back to back on occasion is acceptable

Reply:
i thought it would be useful to demo how to use the keep_alive flag.
i assumed that adding 2 lines of code with comments would be fine... but it seems i was wrong 😆

Reply:
😆 sometimes all it takes is two lines to throw the likes of me astray

It's no big deal, if you think it's better with the demonstration of keep_alive than without, I certainly don't feel strongly enough about it to stop you!